cargo-dist/src/tasks.rs

```diff
@@ -1094,6 +1098,8 @@ impl<'pkg_graph> DistGraphBuilder<'pkg_graph> {
         let system = SystemInfo {
             id: system_id.clone(),
             cargo_version_line,
+            // Real value calculated during the build
+            build_environment: BuildEnvironment::Indeterminate,
```

Comment:
hmm interesting... probably fine...

Reply:
The idea here is this also stands in for "we couldn't calculate it" or "it didn't get calculated for some reason". So set it as the default, and if it lands there on the end we know we had some kind of unmeasurable system (or a logic error)